packages/wallets/backpack/src/adapter.ts

```diff
@@ -0,0 +1,259 @@
+
+            // eslint-disable-next-line @typescript-eslint/no-non-null-assertion
+            const wallet = window!.backpack!;
+
+            if (!wallet.isConnected) {
+                try {
+                    await new Promise<void>((resolve, reject) => {
+                        const handleConnect = () => {
+                            wallet.off('connect', handleConnect);
+                            resolve();
+                        };
+
+                        // Backpack emits a connect event when it is ready
+                        wallet.on('connect', handleConnect);
+
+                        wallet.connect().catch(() => {
+                            wallet.off('connect', handleConnect);
+                            reject(new WalletWindowClosedError());
+                        });
+                    });
+                } catch (error: any) {
+                    if (error instanceof WalletError) throw error;
+                    throw new WalletConnectionError(error?.message, error);
+                }
+            }
+
+            if (!wallet.publicKey) throw new WalletAccountError();
+
+            let publicKey: PublicKey;
+            try {
+                publicKey = new PublicKey(wallet.publicKey.toBytes());
+            } catch (error: any) {
+                throw new WalletPublicKeyError(error?.message, error);
+            }
+            wallet.on('disconnect', this._disconnected);
+
+            this._wallet = wallet;
+            this._publicKey = publicKey;
+
+            this.emit('connect', publicKey);
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        } finally {
+            this._connecting = false;
+        }
+    }
+
+    async disconnect(): Promise<void> {
+        const wallet = this._wallet;
+        if (wallet) {
+            wallet.off('disconnect', this._disconnected);
+
+            this._wallet = null;
+            this._publicKey = null;
+
+            try {
+                await wallet.disconnect();
+            } catch (error: any) {
+                this.emit('error', new WalletDisconnectionError(error?.message, error));
+            }
+        }
+
+        this.emit('disconnect');
+    }
+
+    async sendTransaction(
+        transaction: Transaction,
+        connection?: Connection,
+        options?: SendTransactionOptions
+    ): Promise<TransactionSignature> {
+        try {
+            const wallet = this._wallet;
+            if (!wallet) throw new WalletNotConnectedError();
+
+            const { signers, ...sendOptions } = options ? options : { signers: undefined };
+
+            const resp = await wallet.send(transaction, signers, sendOptions, connection);
+            if (!resp) {
+                throw new WalletWindowClosedError();
+            }
+            return resp;
+        } catch (error: any) {
+            this.emit('error', error);
+            throw error;
+        }
+    }
```

Comment:
The general structure of this method has been updated elsewhere, e.g. https://github.com/solana-labs/wallet-adapter/blob/44a7d3b4910e2ae175ddfef6799a168999fd6ec3/packages/wallets/phantom/src/adapter.ts#L194-L219

Reply:
Is the main thing here that we need to call `prepareTransaction`?

Reply:
It appears that the purpose of `prepareTransaction` https://github.com/solana-labs/wallet-adapter/blob/master/packages/core/base/src/adapter.ts#L91-L99 is to add a fee payer and the recent blockhash. Backpack's `send` implementation does this,  so that we can generate anchor clients without this library.

Reply:
Hmm, the most critical thing is that the method signature matches (connection is required, default param isn't but then you don't need the `options || {}`). The method implementation is discretionary though.

And how do you want to handle partial signers? In general I think it makes sense to sign before passing to the wallet, rather than pass signers to the wallet (since the wallet should be able to handle a transaction arriving with partial signatures anyway). And if the partial signatures should be done in the app, the tx will have to have a feepayer/blockhash.

Reply:
If I understand correctly, I think both use cases are currently covered. The signers here is a convenience to eliminate the `partialSign` boilerplate (and it conforms to the Anchor interface), allowing the user to additionally (and optionally) pass in those signer keys. If you want to partial sign inside the app as you suggest, you can still do so by not providing empty signers.